cloud/http.go

```diff
@@ -141,16 +143,16 @@ func (c *client) doCall(ctx context.Context, method, url string, opts ...request
 	return status, body, callErr
 }
 
-func shouldRetry(status int, body []byte, callErr error, warnFunc func(string, ...interface{})) (bool, error) {
+func shouldRetry(status int, body []byte, callErr error, warnFunc func(string, ...interface{}), reqID string) (bool, error) {
 	if status == http.StatusUnauthorized {
 		return true, nil
 	}
 	if 500 <= status && status <= 599 {
 		msg, err := getMessageFromJSON(bytes.NewReader(body))
 		if err != nil {
-			warnFunc("retrying http request due to unexpected status code %v", status)
+			warnFunc("retrying http request due to unexpected status code %v {reqID: %s}", status, reqID)
```

Comment:
Should `warnFunc` just append the request ID to the message string? It seems like we already pass it.

Reply:
`warnFunc` is passed in to this func here, but is created much higher up in the earthly code before the requestID is created, so I think it might be simpler to append it here

Reply:
Oh ok, it looks like it's already using printf-style arguments.